grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -147,6 +154,26 @@ class PermissionService {
 		return directPermissions
 	}
 
+	private Permission hasTransitiveDashboardPermissions(Canvas canvas, Userish userish) {
+		List<DashboardItem> items = DashboardItem.findAllByCanvas(canvas)
```

Comment:
Result set could theoretically be large, but in practice it's unlikely to be. Would be better to do it all in one query using a join, but that said, I think the current approach will work for now & figuring out how to do it correctly via Hibernate criteria queries might take too much time vs. the benefits.